server/game/gameSystems/DiscardCardsFromHand.ts

```diff
@@ -0,0 +1,84 @@
+import { EventName, Location, RelativePlayer, TargetMode } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import { Card } from '../core/card/Card';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+
+export interface IDiscardCardsFromHandProperties extends IPlayerTargetSystemProperties {
+    amount?: number;
+    cardCondition?: (card: Card, context: AbilityContext) => boolean;
+}
+
+export class DiscardCardsFromHand<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardCardsFromHandProperties> {
+    protected override defaultProperties: IDiscardCardsFromHandProperties = {
+        amount: 1,
+        cardCondition: () => true
```

Comment:
I personally would prefer to default to a type like what we expect, like what happening here. I think this will often lead to less errors and allow us to implement logic a bit easier cuz we don't have check for null constantly. Null can also. Granted once we are in a full TS system the error justification probably is not a problem anymore cuz TS should catch type issues at compile.

But that being said I don't think using TS/JS really adds much to the conversation. Its more up to preference. I think both are fine, and TS will catch issues either way.

Reply:
Sounds good to me!